server/RdtClient.Service/Services/TorrentClients/TorBoxTorrentClient.cs

```diff
@@ -284,39 +284,22 @@ public async Task<Torrent> UpdateData(Torrent torrent, TorrentClientTorrent? tor
         return torrent;
     }
 
-    public async Task<IList<String>?> GetDownloadLinks(Torrent torrent)
+    public async Task<IList<DownloadInfo>?> GetDownloadInfos(Torrent torrent)
     {
         var torrentId = await GetClient().Torrents.GetHashInfoAsync(torrent.Hash, skipCache: true);
 
         return torrent.Files.Where(file => fileFilter.IsDownloadable(torrent, file.Path, file.Bytes))
-                      .Select(file => $"https://torbox.app/fakedl/{torrentId?.Id}/{file.Id}")
+                      .Select(file => new DownloadInfo { RestrictedLink = $"https://torbox.app/fakedl/{torrentId?.Id}/{file.Id}", FileName = Path.GetFileName(file.Path)})
                       .ToList();
     }
 
-    public async Task<String> GetFileName(String downloadUrl)
+    /// <inheritdoc />
+    public Task<String> GetFileName(Download download)
     {
-        if (String.IsNullOrWhiteSpace(downloadUrl))
-        {
-            return "";
-        }
-
-        var uri = new Uri(downloadUrl);
-
-        using (HttpClient client = new())
-        {
-            var request = new HttpRequestMessage(HttpMethod.Head, uri);
-            var response = await client.SendAsync(request);
-            if (response.Content.Headers.ContentDisposition != null)
-            {
-                var fileName = response.Content.Headers.ContentDisposition.FileName;
-                if (!String.IsNullOrWhiteSpace(fileName))
-                {
-                    return fileName.Trim('"');
-                }
-            }
-        }
-
-        return HttpUtility.UrlDecode(uri.Segments.Last());
+        // FileName is set in GetDownlaadInfos
```

Comment:
so does this eliminate the HEAD request to the unrestricted download url?

Reply:
Okay, looks good to me, good job on getting rid of that request i spent ages trying to think of how to get around doing it but now that I see the solution it seems so obvious haha